zcash_client_sqlite/src/wallet.rs

```diff
@@ -245,6 +248,43 @@ impl ViewingKey {
     }
 }
 
+pub(crate) fn seed_matches_derived_account<P: consensus::Parameters>(
+    params: &P,
+    seed: &SecretVec<u8>,
+    seed_fingerprint: &SeedFingerprint,
```

Comment:
I find it surprising that this takes both the seed and the seed fingerprint, but I guess that since it's a helper it's fine.

Reply:
These are from two different data sources:
- `seed` comes from the caller.
- `seed_fingerprint` comes from the wallet.

The helper is explicitly checking their consistency.